### ISSUE TITLE
Create plot themes

### ISSUE THREAD

**Issue by mkfreeman** [observablehq/plot#630]
To support different styles, we could create a set of plot themes (e.g., `dark`, `light`, etc.). As a comparison, the `ggthemes` [package](https://mran.microsoft.com/snapshot/2017-02-04/web/packages/ggthemes/vignettes/ggthemes.html) has default styles to emulate popular styles like the Wallstreet Journal, FiveThirtyEight, etc. 

Perhaps a simple approach would be to define a set of styles in `Plot.themes` that could be set as the `style` property in a Plot:

```js
Plot.plot({
    style: Plot.themes.dark
})
```

Where `Plot.themes.dark` has a set of key/value pairs describing the style: `{background: "black", color: "white",...}`

**Comment by Fil** [comment]
related: https://github.com/observablehq/plot/issues/114

**Comment by awiederkehr** [comment]
Would love to help work on this however I can!

**Comment by mbostock** [comment]
The new axis mark #1197 will help with this, for example you could have a ggplot2 theme https://github.com/observablehq/plot/pull/1197#issuecomment-1364573933.

**Comment by Fil** [comment]
Here's a handy response from Mike on the user forum:
https://talk.observablehq.com/t/set-global-plot-defaults-for-observable-plot/7797/3

**Comment by martindaniel4** [comment]
As a Plot user, that feature would be very useful! Here is also another R package that was relevant to style graphs from a former Airbnb DS - https://github.com/ricardo-bion/ggtech

**Comment by tophtucker** [comment]
I like Mike’s nice `style: Plot.themes.dark` suggestion. And it could be prototyped as an ongoing notebook.

It gets trickier when you want to set a default theme across many plots. One can also imagine wanting to set default categorical/sequential/diverging color schemes, etc.

Maybe there could be something like 

```
Plot.defaults({
  style: Plot.themes.dark,
  color: {scheme: {sequential: "blues"}}
})
```

which would return a new instance of Plot with those defaults? There’d be various difficulties. What’s that object syntax? (Note that I abused the color scheme option.) To name your version Plot in a notebook, would you have to do something like `Plot = (await require("@observablehq/plot")).defaults({style: Plot.themes.dark})`, or could you take advantage of the built-in Plot?

- https://ggplot2.tidyverse.org/reference/theme_get.html
- https://vega.github.io/vega-lite/docs/config.html

**Comment by mbostock** [comment]
Yes, exactly Toph. On Observable, my hope is that you can configure Plot for your workspace and it automatically applies the theme to notebooks in your workspace without you having to edit them. And likewise in vanilla JavaScript, you could call `export default Plot.defaults(…)` or whatever and import that instead of Plot.

**Comment by mbostock** [comment]
> Projections!

What about projections, @Fil? Not sure how this is related to themes.

**Comment by Fil** [comment]
In that [comment](https://github.com/observablehq/plot/pull/1111#issuecomment-1327609392) I wanted a mechanism to add a few named projections to my "themed" or personalized Plot, similarly to the way we'll be able to add (or modify) color themes. Once we have a theming mechanism in place it will seem natural to want to extend it (progressively) to other hard-coded lists of named option values. Even more so if it's called "defaults" and not "theme".

**Comment by christophe-g** [comment]
Stumbled across this while trying to adjust my observable colors to my themes. 

What about adding a new scheme for categorical color space along the following, using css variables: 
```ts
export default [
  'var(--plot-color-0)',
  'var(--plot-color-1)',
  'var(--plot-color-2)',
  'var(--plot-color-3)',
  'var(--plot-color-4)',
  'var(--plot-color-5)',
  'var(--plot-color-6)',
  'var(--plot-color-7)',
  'var(--plot-color-8)',
  'var(--plot-color-9)',
]
``` 
It would then be the consumer responsibility to set relevant colors - we could provide a helper so that it uses `observable10` scheme by default. Handling dark, and customized theme is then pretty easy.

Happy to provide a PR if that makes sense

**Comment by Fil** [comment]
@christophe-g I like the idea of this virtual color scheme, but unfortunately I think this can't be the default because for the **raster** mark we need to convert the css colors into materialized RGB colors.

**Comment by christophe-g** [comment]
Merci Fil, 

>  for the raster mark we need to convert the css colors into materialized RGB colors

I guess you are referring to https://github.com/observablehq/plot/blob/7e09c31379acb3a3e606c1364f22674727ced35f/src/marks/raster.js#L145

I think this type of virtual color scheme should only be applied/available for categorical data, not for continuous scales. I am thinking of this approach as a generalization of `observable10` scheme for instance.

In theory, when color processing is required (like compute rgb from a real color and not a css variable), it should be possible to infer the value of the css variable in a given context though: 

```
// Get the root element - the problem is that this is not going to work well with shadow-dom, 
// you would need the root of the web-component for which css variables are defined. 
const root = document.documentElement;
const style = getComputedStyle(root);
const color = style.getPropertyValue('--plot-color-0);
```

**Comment by Fil** [comment]
This would work in many cases, but it's tricky to do generically. It's possible to use categorical colors in the **raster** mark, and we can't know in advance where in the DOM the chart might be appended. (Also, the materialized values for a variable might change with time, for example when entering dark mode.)

But outside of this caveat, it seems quite useful to have as a categorical palette.
